packages/wallet-protocols/src/protocols/create-and-direct-fund/protocol.ts

```diff
@@ -0,0 +1,120 @@
+import { Machine, MachineConfig, sendParent } from 'xstate';
+
+import { JsonRpcCreateChannelParams } from '../../json-rpc';
+import { MachineFactory } from '../../machine-utils';
+import { Store, success } from '../..';
+
+import { ethAllocationOutcome } from '../../calculations';
+
+import { AdvanceChannel, DirectFunding } from '..';
+
+const PROTOCOL = 'create-and-direct-fund';
+
+export enum Indices {
+  Left = 0,
+  Right = 0,
+}
+
+export type Init = JsonRpcCreateChannelParams & {
+  channelId: string;
+  challengeDuration: number;
+  index: Indices;
+};
+
+export type FirstStateConstructed = Init & { type: 'FIRST_STATE_CONSTRUCTED' };
+
+export const advanceChannelArgs = (i: 1 | 3) => ({ channelId }: Init): AdvanceChannel.Init => ({
+  channelId,
+  targetTurnNum: i,
+});
+
+const constructFirstState = {
+  invoke: {
+    id: 'constructFirstState',
+    src: 'constructFirstState',
+    data: (ctx: Init) => ctx,
+    onDone: 'preFundSetup',
+  },
+};
+
+const preFundSetup = {
+  invoke: {
+    id: 'preFundSetup',
+    src: 'advanceChannel',
+    data: advanceChannelArgs(1),
+    onDone: 'funding',
+  },
+  on: { CHANNEL_CLOSED: 'abort' },
+};
+
+// FIXME: Abort should not be success
+const abort = success;
+
+const directFunding = {
+  invoke: {
+    src: 'direct-funding',
+    data: ({ allocations, channelId }: FirstStateConstructed): DirectFunding.Init => {
+      return {
+        channelId,
+        minimalAllocation: allocations,
+      };
+    },
+    onDone: 'postFundSetup',
+  },
+};
+
+const postFundSetup = {
+  invoke: {
+    id: 'postFundSetup',
+    src: 'advanceChannel',
+    data: advanceChannelArgs(3),
+    onDone: 'success',
+  },
+};
+
+type Context = Init;
+export const config: MachineConfig<Context, any, any> = {
+  key: PROTOCOL,
+  initial: 'constructFirstState',
+  states: {
+    constructFirstState,
+    preFundSetup,
+    abort,
+    funding: directFunding,
+    postFundSetup,
+    success: {
+      type: 'final' as 'final',
+
+      entry: sendParent({ type: 'CHANNEL_CREATED' }),
```

Comment:
I don't think we need to send `CHANNEL_CREATED` to the parent anymore. Likewise, the channelId is already known to the parent -- it supplied it as the `Init` args -- so it doesn't need to be returned in the next line.